### ISSUE TITLE
gaps - inner gaps disappear from tabbed/stacked containers  

### ISSUE THREAD

**Issue by budRich** [i3/i3#5261]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

inner gaps disappear from tabbed/stacked containers  

This one is hard to describe in text, but i recorded a short demonstration:

https://user-images.githubusercontent.com/2143465/200381894-16ac6de2-bc9e-4db9-8efb-10da6cbf54e8.mp4

this layout:  
```text
workspace with splith layout
  leaf window A
  leaf window B
  tabbed container
    leaf window C
    leaf window D
```

these settings:
`gaps inner 10`

![splits1](https://user-images.githubusercontent.com/2143465/200383351-3a9b1496-b0d9-4f5c-9af2-43908b3fe93f.png)

everything looks and works as expected till you change the layout to something like this:  

```text
workspace with splitv layout
  splith container
    leaf window A
    tabbed container
      leaf window C
      leaf window D
  leaf window B
```

![split2](https://user-images.githubusercontent.com/2143465/200383392-e9eca038-050f-4af9-b697-24c26f6287eb.png)

the inner gaps around the tabbed container are now removed. adjusting the layout like this is one single tile-drag op .

## Expected Behavior

gaps are not removed

## Reproduction Instructions

see description of expected behavior 

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.21-113-ge6a28b94 © 2009 Michael Stapelberg and contributors
Running i3 version: 4.21-113-ge6a28b94 (pid 716793)
...

</pre>

<details><summary>Config file</summary><pre>
gaps inner 10
</pre>
</details>

<pre>
- Linux Distribution & Version: Ubuntu Lunar Lobster (development branch)
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

**Comment by i3bot** [comment]
I don’t see a link to logs.i3wm.org. Did you follow https://i3wm.org/docs/debugging.html? (In case you actually provided a link to a logfile, please ignore me.)

**Comment by bisby** [comment]
This was actually the way I was able to tell that I was successfully changed to i3-next instead of i3-gaps.

I'm experiencing the same thing demonstrated by OP... tabs seem fine except when they are nested.
The left side of my split is fine tabbed or split. The right side of my split has a nested vertical split... and the top/bottom of that split is where issues happen, and when tabbed they will override the horizontal split's gaps even.

I tried with smart_gaps on and off, which seemed to make no difference, and I'm running from `i3-git` from the AUR
